### PR TITLE
Translation(PT-BR): Nullable Types.ts

### PR DESCRIPTION
**Pull request by YPAzevedo**
Add PT-BR translations to:

- Nullable Types.ts file from Primitive Types

### CODE REVIEW

**Review by fpuddu — APPROVED** [review]

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+//// { order: 3, compiler: { strictNullChecks: false } }
+
+// No JavaScript existem duas maneiras de declarar valores que não
+// existem, o TypesScript adiciona sintaxe extra que nos proporciona
+// mais maneiras de declarar algo como opicional ou nulo.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// mais maneiras de declarar algo como opcional ou nulo.
```

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  audio: { name: "LEGO Movie Soundtrack" },
+};
+
+// Por que não usar undefined? Pricipalmente, porque agora podemos
+// verificar que text foi incluido corretamente. Se text retornasse
+// undefined então o resultado seria o mesmo que dizer que não
+// foi declarado.
+
+// Isso pode parecer superficial, mas quando convertido
+// em uma JSON string, se o text fosse undefined, nāo seria
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// em uma string JSON, se o text fosse undefined, nāo seria
```

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+// incluido na conversão para string.
+
+// Strict Null Types
+
+// Antes do TypeScrips 2.0, undefined e null eram efetivamente
+// ignorados no sistema de typagem. Isso fazia com que TypeScript
+// pudesse prover um ambiente de codagem mais próximo de um JavaScript
+// sem tipagem.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// Antes do TypeScript 2.0, undefined e null eram efetivamente
// ignorados no sistema de tipagem. Isso fazia com que TypeScript
// pudesse prover um ambiente de desenvolvimento mais próximo de
// um JavaScript sem tipagem.
```

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+// A versāo 2.0 adiciona um marcador de compilagem, chamado "strictNullChecks"
+// esse marcador requer que undefined e null fossem tratados como tipos,
+// o que significa que devem ser manipulados via análise de code-flow.
+// ( veja mais em example:code-flow )
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// A versāo 2.0 adiciona uma configuração de compilação (flag),
// chamada "strictNullChecks". Essa flag requer que undefined e null
// sejam tratados como tipos, o que significa que devem ser manipulados
// via análise de fluxo de código (code-flow).
// ( veja mais em example:code-flow )
```

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+// Para um exemplo da diferença quando se usa o stric null
+// checks no TypeScript, passe o mouse por cima de "Potential String" abaixo:
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// Para um exemplo da diferença quando se usa o strict null
// checks no TypeScript, passe o mouse por cima de "Potential String" abaixo:
```

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+type PotentialString = string | undefined | null;
+
+// O PotentialString discarta o undefined e null. Se você
+// até as configurações, ligar o strict mode e voltar
+// você pode ver que ao passar o mouse por cime de PotencialString
+// agora pode ver a união completa.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// O PotentialString discarta o undefined e null. Se você ir
// até as configurações, ligar o strict mode e voltar
// você pode ver que ao passar o mouse por cima de PotencialString
// agora pode ver a união completa.
```

**Review by bosi-programming — APPROVED** [review]

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+declare function getID(): PotentialString;
+
+const userID = getID();
+console.log("User Logged in: ", userID.toUpperCase());
+
+// Somente em strict mode a linha acima vai falhar^
+
+// Existem maneiras de dizer ao TypeScript que você melhor sobre a tipagem,
+// como type assertion ou através do non-null assertion operator (!)
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// Existem maneiras de dizer ao TypeScript que você sabe mais
// sobre a tipagem, como type assertion ou através do
// non-null assertion operator (!)
```

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+// Void é a tipagem de retorno quando uma função não
+// tem um valor de retorno.
+
+const voidFunction = () => {};
+const resultOfVoidFunction = voidFunction();
+
+// Isso geralmente é um acidente, e o TypeScript mantém o tipo void
+// para que você tenha erros de compilação - mesmo que em
+// runtime ele seria um valor undefined.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// para que você tenha erros de compilação - mesmo que em
// tempo de execução ele seria um valor undefined.
```

**Review by danilofuchs — CHANGES_REQUESTED** [review]
Great job!
I found some typos and some phrases we can rework to increase clarity. Thanks!

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts (hunk 4 above)
Comment (YPAzevedo) [reply packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]: Yes, some words are harder to find a good translation for. Thanks

**Comment by YPAzevedo** [comment]
@danicole thanks for the inputs, accepted the suggestions. I need to do some proof reading next time hehe. and some words are hard to translate.

File: packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
+//// { order: 3, compiler: { strictNullChecks: false } }
+
+// No JavaScript existem duas maneiras de declarar valores que não
+// existem, o TypesScript adiciona sintaxe extra que nos proporciona
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Type Primitives/Nullable Types.ts]:
```suggestion
// existem, o TypeScript adiciona sintaxe extra que nos proporciona
```

**Review by danilofuchs — CHANGES_REQUESTED** [review]
Sorry, I missed one 🙈

**Comment by danilofuchs** [comment]
> thanks for the inputs, accepted the suggestions. I need to do some proof reading next time hehe. and some words are hard to translate.

No worries!
Everyone makes some mistakes when translating. That's why we follow this PR review process!

LGTM

**Pull request merged**